packages/core/test/unit/application.test.ts

```diff
@@ -8,10 +8,60 @@ import {Application, Server, Component} from '../../index';
 import {Context, Constructor} from '@loopback/context';
 
 describe('Application', () => {
+  describe('controller binding', () => {
+    let app: Application;
+    class MyController {}
+
+    beforeEach(givenApp);
+
+    it('binds a controller', () => {
+      const binding = app.controller(MyController);
+      expect(binding.tags.has('controller')).to.be.true();
+      expect(binding.key).to.equal('controllers.MyController');
+      expect(app.findByTag('controller')[0]).to.equal(binding);
+    });
+
+    it('binds a controller with custom name', () => {
+      const binding = app.controller(MyController, 'my-controller');
+      expect(binding.tags.has('controller')).to.be.true();
+      expect(binding.key).to.equal('controllers.my-controller');
+      expect(app.findByTag('controller')[0]).to.equal(binding);
+    });
+
+    function givenApp() {
+      app = new Application();
+    }
+  });
+
+  describe('component binding', () => {
+    let app: Application;
+    class MyController {}
+    class MyComponent implements Component {
+      controllers = [MyController];
+    }
+
+    beforeEach(givenApp);
+
+    it('binds a component', () => {
+      app.component(MyComponent);
+      expect(app.findByTag('component').length).to.equal(1);
```

Comment:
If this assertion fails, the error message will be most likely not helpful. E.g. if the length is 2, then we cannot tell what are the other bindings.


I am proposing the following improvement:

```ts
expect(app.findByTag('component').map(b => b.key))
  .to.containEql('components.my-component');
```

To avoid repetition, we should extract `app.findByTag('component').map(b => b.key)` into a shared helper, e.g.

```ts
function findKeysByTag(ctx, tag) {
  return ctx.findByTag(tag).map(b => b.key)
}
```